lib/Dialect/TTNN/IR/TTNNWorkaroundInterface.cpp

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "ttmlir/Dialect/TTNN/IR/TTNNWorkaroundInterface.h"
+#include "ttmlir/Dialect/TTNN/IR/TTNNOps.h"
+#include "ttmlir/Dialect/TTNN/IR/TTNNWorkarounds.h"
+#include <mlir/Interfaces/DestinationStyleOpInterface.h>
+
+namespace mlir::tt::ttnn {
+#include "ttmlir/Dialect/TTNN/IR/TTNNWorkaroundInterface.cpp.inc"
+
+// Verifier function for TTNN Workaround Interface
+mlir::LogicalResult verifyTTNNWorkaroundInterface(mlir::Operation *op) {
+  TTNNWorkaroundInterface workaroundOp =
+      mlir::cast<TTNNWorkaroundInterface>(op);
+
+  // Verify that the number of input and output operand workarounds is the same
+  // as the number of tensor operands and tensor results
+  size_t tensorInputs = 0;
+  size_t tensorResults = 0;
+
+  // Count the number of tensor input operands including DPS inits
+  for (auto operand : op->getOperands()) {
+    if (mlir::isa<::mlir::RankedTensorType>(operand.getType())) {
```

Comment:
Just wondering why do we check this, isn't every operand of RankedTensorType?

Reply:
Nope, there can be ops that for example accepts device as an operand. For example, in TTNN for empty op, we have something like this:
%0 = "ttnn.get_device"() <{mesh_shape = #ttnn<mesh_shape 1x1>}> : () -> !tt.device<#device>
...
%2 = "ttnn.empty"(%0) <{dtype = #tt.supportedDataTypes<f32>, layout = #ttnn.layout<row_major>, memory_config = #ttnn.memory_config<<interleaved>, #dram, <<64x128>>>, shape = #ttnn.shape<64x128>}> : (!tt.device<#device>) -> tensor<64x128xf32, #ttnn_layout1>
...

Reply:
Ah I see, thanks.